server/server.go

```diff
@@ -711,3 +715,53 @@ func (s *server) RoundRobin(rrs []dns.RR) {
 	}
 
 }
+
+// dedup will de-duplicate a message on a per section basis.
+// Multiple identical (same name, class, type and rdata) RRs will be coalesced into one.
+func dedup(m *dns.Msg) *dns.Msg {
+	// Answer section
+	ma := make(map[string]dns.RR)
+	for _, a := range m.Answer {
+		// Or use Pack()... Think this function also could be places in go dns.
+		s := a.Header().Name
+		s += strconv.Itoa(int(a.Header().Class))
+		s += strconv.Itoa(int(a.Header().Rrtype))
+		for i := 1; i <= dns.NumField(a); i++ {
+			s += dns.Field(a, i)
+		}
+		ma[s] = a
+	}
+	// Only is our map is smaller than the #RR in the answer section we should reset the RRs
+	// in the section it self
+	if len(ma) < len(m.Answer) {
+		i := 0
+		for _, v := range ma {
+			m.Answer[i] = v
+			i++
+		}
+		m.Answer = m.Answer[:len(ma)]
```

Comment:
Does this handle the case of multiple multiples?  Seems like it is worth a testcase

```
# dig @10.0.0.10 +noall +answer +additional +ignore svc.ns.kubernetes.local A
svc.ns.kubernetes.local. 3600	IN	A	3.3.3.3
svc.ns.kubernetes.local. 3600	IN	A	2.2.2.2
svc.ns.kubernetes.local. 3600	IN	A	3.3.3.3
svc.ns.kubernetes.local. 3600	IN	A	2.2.2.2
svc.ns.kubernetes.local. 3600	IN	A	1.1.1.1
svc.ns.kubernetes.local. 3600	IN	A	1.1.1.1
```